packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -418,6 +423,18 @@ contract Colosseum is Initializable, Semver {
         emit Proven(_outputIndex, _outputRoot);
     }
 
+    /**
+     * @notice Reverts if the output of given index is already finalized.
+     *
+     * @param _outputIndex Index of the L2 checkpoint output.
+     */
+    function _checkOutputNotFinalized(uint256 _outputIndex) private view {
```

Comment:
What about changing the function to a modifier?

Reply:
I applied it in last force-push, please check it again! (I also changed the order of some functions in `guardian.go`.)